draft-ietf-protected-audience-key-value-services.md

```diff
@@ -318,9 +318,54 @@ Each key group is expected to have exactly one tag from the following list:
 | renderUrls | "renderUrls" represent URLs for advertisements to be looked up from the service's internal datastore. |
 | adComponentRenderUrls | "adComponentRenderUrls" represent component URLs for advertisements to be looked up from the service's internal datastore. |
 
-### Generating a Request
-
-TODO
+### Generating a Request {#request-generate}
+
+This section describes how the client MAY form and serialize request
+messages in order to fetch values from the Trusted Key Value server.
+
+This algorithm takes as input an [HPKE] `public key` and its associated
+`key id`, a `metadata` map for global configuration, where both keys and
+values are strings, and `compression groups`, which is a list of
+compression group structs. Each struct contains a compression group with
+a compression group ID and one or more partitions. Each partition
```

Comment:
```suggestion
a `compressionGroupID` and one or more `partition`s. Each `partition`
```
I'm not sure if I'm using the right quotes, but I think we need to make this match the steps below

Reply:
Changed to `compression group id` which represents the value of partition["compressionGroupId"].